app/main.py

```diff
@@ -1 +1,40 @@
-# write your code here
+class Animal:
+
+    alive = []
+
+    def __init__(self,
+                 name: str,
+                 health: int = 100,
+                 hidden: bool = False
+                 ) -> None:
+        self.health = health
+        self.hidden = hidden
+        self.name = name
+        Animal.alive.append(self)
+
+    def __repr__(self) -> str:
+        return (
+            f"{{Name: {self.name}, "
+            f"Health: {self.health}, "
+            f"Hidden: {self.hidden}}}"
+        )
+
+
+class Herbivore(Animal):
+    def hide(self) -> None:
+        if self.hidden:
+            self.hidden = False
+        else:
+            self.hidden = True
+
+
+class Carnivore(Animal):
+    def bite(self, animal: Herbivore) -> Animal:
+        if animal.hidden or isinstance(animal, Carnivore):
```

Comment:
```suggestion
        if not animal.hidden or isinstance(animal, Carnivore):
```
And than do some operations, remove redundant code